packages/ui-date-time-input/src/DateTimeInput/index.tsx

```diff
@@ -0,0 +1,559 @@
+
+  handleRef = (el: Element | null) => {
+    this.ref = el
+  }
+
+  constructor(props: DateTimeInputProps) {
+    super(props)
+    // State needs to be calculated because render could be called before
+    // componentDidMount()
+    this.state = this.recalculateState(props.value || props.defaultValue)
+  }
+
+  componentDidMount() {
+    // we'll need to recalculate the state because the context value is
+    // set at this point (and it might change locale & timezone)
+    const initState = this.recalculateState(
+      this.props.value || this.props.defaultValue
+    )
+    this.setState(initState)
+  }
+
+  componentDidUpdate(prevProps: Readonly<DateTimeInputProps>): void {
+    const valueChanged =
+      prevProps.value !== this.props.value ||
+      prevProps.defaultValue !== this.props.defaultValue
+    const isUpdated =
+      valueChanged ||
+      prevProps.locale !== this.props.locale ||
+      prevProps.timezone !== this.props.timezone ||
+      prevProps.dateFormat !== this.props.dateFormat ||
+      prevProps.messageFormat !== this.props.messageFormat ||
+      prevProps.invalidDateTimeMessage !== this.props.invalidDateTimeMessage
+
+    if (isUpdated) {
+      this.setState((_prevState: DateTimeInputState) => {
+        return {
+          ...this.recalculateState(this.props.value || this.props.defaultValue)
+        }
+      })
+    }
+  }
+
+  recalculateState(
+    dateStr?: string,
+    doNotChangeDate = false,
+    doNotChangeTime = false
+  ): DateTimeInputState {
+    let errorMsg: FormMessage | undefined
+    if (dateStr) {
+      const parsed = DateTime.parse(dateStr, this.locale(), this.timezone())
+      if (parsed.isValid()) {
+        if (doNotChangeTime && this.state.timeSelectValue) {
+          // There is a selected time, adjust the parsed date to its value
+          const timeParsed = DateTime.parse(
+            this.state.timeSelectValue,
+            this.locale(),
+            this.timezone()
+          )
+          parsed.hour(timeParsed.hour()).minute(timeParsed.minute())
+        }
+        if (doNotChangeDate && this.state.iso) {
+          parsed
+            .date(this.state.iso.date())
+            .month(this.state.iso.month())
+            .year(this.state.iso.year())
+        }
+        const newTimeSelectValue = this.state?.timeSelectValue
+          ? this.state.timeSelectValue
+          : this._timeInput
+              ?.getBaseDate()
+              .minute(parsed.minute())
+              .hour(parsed.hour())
+              .toISOString()
+        return {
+          iso: parsed.clone(),
+          calendarSelectedDate: parsed.clone(),
+          dateInputTextChanged: false,
+          dateInputText: parsed.format(this.dateFormat),
+          message: {
+            type: 'success',
+            text: this.props.messageFormat
+              ? parsed.format(this.props.messageFormat)
+              : parsed.format(DateTimeInput.DEFAULT_MESSAGE_FORMAT)
```

Comment:
Well, the `defaultProps` will catch it if it is `undefined`. In case of `null`... hmm, not sure I'd write a guard for that, it is the users fault if they set it to `null`... Not to mention that it will be underlined because of the TS type.

Reply:
This is a safeguard for the case if someone sets it to undefined/null after creation (e.g. by accident)

Reply:
removed it